### DISCUSSION TITLE
Version detection question

### DISCUSSION THREAD

**Discussion by jelockwood** [autopkg/autopkg#938] [General]
In the Mac world at least it is more common for version numbers of software to use a format like 1.0, 1.1, 1.2.1, etc.

A tool for which I have previously written an AutoPkg recipe called Smartmontools however uses a format like 7.4, and 7.4-1

This has resulted in my recipe treating _both_ as 7.4 with the -1 portion being skipped.

Whilst arguably the developer is using the wrong format :) does anyone have a suggestion for how to handle this in a set of AutoPkg recipies?

See - https://github.com/jelockwood/recipes/tree/master/SMART%20Monitoring
and - https://sourceforge.net/projects/smartmontools/files/

**Comment by macmule** [comment]
Isn’t this more of a Munki issue?

And, you’re likely hitting something like: https://github.com/munki/munki/wiki/FAQ#q-i-keep-seeing-warnings-like-warning-could-not-process-item-office2011_update-1442-for-update-no-pkginfo-found-in-catalogs-production-yet-there-is-definitely-an-item-named-office2011_update-1442-in-the-production-catalog-what-is-happening

Regards,

Ben.


[Jamf]<https://www.jamf.com/>


Ben Toms
Principal Software Engineer
Remote, UK
[Phone]
0800 368 9330
[Email]
***@***.******@***.***>
[Web]
www.jamf.com<https://www.jamf.com/>
[Facebook]<https://www.facebook.com/jamfsoftware/>   [Twitter] <https://twitter.com/JamfSoftware>    [LinkedIn] <https://www.linkedin.com/company/jamf-software>    [Jamf YouTube] <https://www.youtube.com/user/jamfmedia>


From: jelockwood ***@***.***>
Date: Wednesday, 17 April 2024 at 10:01
To: autopkg/autopkg ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [autopkg/autopkg] Version detection question (Discussion #938)

In the Mac world at least it is more common for version numbers of software to use a format like 1.0, 1.1, 1.2.1, etc.

A tool for which I have previously written an AutoPkg recipe called Smartmontools however uses a format like 7.4, and 7.4-1

This has resulted in my recipe treating both as 7.4 with the -1 portion being skipped.

Whilst arguably the developer is using the wrong format :) does anyone have a suggestion for how to handle this in a set of AutoPkg recipies?

See - https://github.com/jelockwood/recipes/tree/master/SMART%20Monitoring
and - https://sourceforge.net/projects/smartmontools/files/

**Comment by gregneagle** [comment]
It's not a Munki issue:
```
$ munki-python 
Python 3.12.2 (v3.12.2:6abddd9f6a, Feb  6 2024, 17:02:06) [Clang 13.0.0 (clang-1300.0.29.30)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from munkilib.pkgutils import MunkiLooseVersion
>>> MunkiLooseVersion("7.4") < MunkiLooseVersion("7.4-1")
True
```
It's more likely he's describing the issue with the PkgCreator processor that is quite opinionated about what are legal characters in a package version, but I'd invite @jelockwood to elaborate.

**Reply by jelockwood** [reply]
Thanks everyone for your responses.

The file name of the installer as mentioned has a version number in it of 7.4-1 to differentiate it from the previous 7.4 release. I had presumed that the installer pkg _internally_ had this same version format and that this was the problem. This particular developer is frequently using -1 in the file names of releases.

I have however now 'de-compiled' the installer pkg and it appears based on the PackageInfo file that the version number in that is actually also 7.4 and in theory should be 7.4-1

This would therefore explain why my AutoPkg and Munki recipes are listing it as 7.4 and this is presumed the same as the previous version.

I have filed an issue with the developers on the Smartmontools ticket site as this is sadly an even more common type of error made when building installer pkgs.

I am not sure even the power of Munki to check what version has been installed will help since the installer is not installing any apps or bundles or similar to look at their version information. As it stands the Munki pkginfo is being (mis)labelled as version 7.4. I can manually edit the pkginfo and believe that should act as a workaround but that somewhat defeats the point of using AutoPkg.
